lib/src/builder_implementation.dart

```diff
@@ -5125,37 +5261,50 @@ Iterable<DartObject> _getEvaluatedMetadata(
       (ElementAnnotation annotation) => _getEvaluatedMetadatum(annotation));
 }
 
+/// Determine whether the given library is a platform library.
+///
+/// TODO(eernst): This function is only needed until a solution is found to the
+/// problem that platform libraries are considered invalid when obtained from
+/// `getResolvedLibraryByElement`, such that subsequent use will throw.
+/// Issue 173.
+bool _isPlatformLibrary(LibraryElement libraryElement) =>
+    libraryElement.source.uri.scheme == "dart";
+
 /// Adds a severe error to the log, using the source code location of `target`
 /// to identify the relevant location where the error occurs.
-void _severe(String message, [Element target]) {
+Future<void> _severe(String message, [Element target]) async {
   if (target != null) {
-    log.severe(_formatDiagnosticMessage(message, target));
+    log.severe(await _formatDiagnosticMessage(message, target));
   } else {
     log.severe(message);
   }
 }
 
 /// Adds a 'fine' message to the log, using the source code location of `target`
 /// to identify the relevant location where the issue occurs.
-void _fine(String message, [Element target]) {
+Future<void> _fine(String message, [Element target]) async {
   if (target != null) {
-    log.fine(_formatDiagnosticMessage(message, target));
+    log.fine(await _formatDiagnosticMessage(message, target));
   } else {
     log.fine(message);
   }
 }
 
 /// Returns a string containing the given [message] and identifying the
 /// associated source code location as the location of the given [target].
-String _formatDiagnosticMessage(String message, Element target) {
+Future<String> _formatDiagnosticMessage(String message, Element target) async {
```

Comment:
As you suggested offline, it might make things simpler if you gather information for all log messages synchronously and make the async work of getting locations after the main computation.

Reply:
Also covered by https://github.com/dart-lang/reflectable/issues/182.